java/arcs/core/host/AnnotationBasedHostRegistry.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.host
+
+import arcs.core.sdk.Particle
+import kotlin.reflect.KClass
+
+/**
+ * A HostRegistry that automatically registers particles with hosts by matching [TargetHost]
+ * annotated annotations on [Particle] classes with [ArcHost]s.
+ */
+open class AnnotationBasedHostRegistry : HostRegistry {
+    val hosts: MutableList<ArcHost> = mutableListOf()
```

Comment:
Does this need to be public? Would protected work better?

Reply:
Additionally: consider making it just a `List<ArcHost>`, and using a backing property for the mutable variant, so subclasses can't modify it directly (without using `registerHost()`/`unregisterHost()`)